### PR TITLE
Set local trust with Crypto V2 and deprecate legacy verification method

### PR DESCRIPTION
**Pull request by Anderas**
A few final bits to implement CryptoV2:

- set local trust (incl (un)blacklisting users)
- remove deprecated and legacy verification method that does not begin with verification request (use `requestVerificationWithDevice` instead)
- determine room encryption state
- retry decryption of events after manually importing keys

### CODE REVIEW

File: MatrixSDK/Crypto/Verification/MXKeyVerificationManagerV2.swift [hunk 1]
```diff
@@ -399,6 +369,25 @@ class MXKeyVerificationManagerV2: NSObject, MXKeyVerificationManager {
     
     // MARK: - Verification requests
     
+    func requestVerificationByToDevice(
+        withUserId userId: String,
```
Comment (stefanceriu) [inline MatrixSDK/Crypto/Verification/MXKeyVerificationManagerV2.swift:373]: Is this `userId` supposed to be used below instead of the session one? 🤔

**Review by stefanceriu — APPROVED** [review]
Apart from one small comment lgtm 👍

File: MatrixSDK/Crypto/Verification/MXKeyVerificationManagerV2.swift (hunk 1 above)
Comment (Anderas) [reply MatrixSDK/Crypto/Verification/MXKeyVerificationManagerV2.swift:373]: Not sure I understand the question. The method serves to verify any user, incl self, but the ffi  interface has explicitely `requestSelfVerification` vs `requestVerification(user:...)`. So if the passed-in userId matches the session, we use self-verification.

Comment (stefanceriu) [reply MatrixSDK/Crypto/Verification/MXKeyVerificationManagerV2.swift:373]: Ups sorry, my bad. I completely missread that being as an `if **let** userId == session?.myUserId`, please disregard.

**Pull request merged**